File: Kotlin/26.remove_duplicates_from_sorted_array.kt
fun main() {
    println(removeDuplicates(intArrayOf(1,1,2)));
    println(removeDuplicates(intArrayOf(0,0,1,1,1,2,2,3,3,4)));
}

fun removeDuplicates(nums: IntArray): Int {
    val set = nums.toSet()

    set.forEachIndexed { i, v ->
        nums[i] = v;
    }

    return set.size;
}

File: lib/26.remove_duplicates_from_sorted_array.dart
void main() {
  print(removeDuplicates([1, 1, 2]));
  print(removeDuplicates([0, 0, 1, 1, 1, 2, 2, 3, 3, 4]));
}

int removeDuplicates(List<int> nums) {
  final set = nums.toSet();

  var index = 0;
  set.forEach((e) {
    nums[index++] = e;
  });

  return set.length;
}
